docs/user_data_deletion_requests.md

```diff
@@ -0,0 +1,62 @@
+---
+title: (Coming Soon) User Data Deletion Requests API
+---
+
+:::info
+NOTE: This is not quite rolled out yet! We are just sharing this as a preview of what it will look like. If you have feedback or feature requests, please let us know.
+:::
+
+GDPR and similar laws may require that you delete a user's data when they request it. To support deletion requests on our side, we have built this API for you to call.
+
+### How to use the API
+
+All requests must include the STATSIG-API-KEY field in the header. The value should be a Console API Key which can be created in the Project Settings on console.statsig.com/api_keys.
+
+## Sending Requests
+
+Data deletion requests take the following parameters:
+- `unit_type`: The unit type that corresponds to the IDs that you would like to deleted. This could be userID, stableID, or some custom ID that you have set up
+- `ids`: A list of strings, that are the IDs you would like to delete data for
+- `request_id` (optional): If you store a request ID on your side that you would like us to use, you can pass it in, and it will be used to track the request. If left unset, we will provide an ID for you to use in the response. Note that it must be a unique ID if you are supplying your own, otherwise the request will fail with a 400 code.
+
+```bash
+curl \
+  --header "statsig-api-key: <YOUR-API-KEY>" \
```

Comment:
This should probably be under console API, right?  And use a console api key?  This doesnt spell out which api key to use here.

cc @jacob-statsig  RE console api

Reply:
Whoops just reread and saw that it does require a console api key.  This should be documented under `docs/console-api/` then

Reply:
cc @marcos-statsig for what you think about APIs like this and mark_data_ready which are sort of console APIs but also not really console APIs. assuming we will eventually have some docs for that one too right?